server/frontend.go

```diff
@@ -170,71 +225,103 @@ func (s *server) NVMeNamespaceDelete(ctx context.Context, in *pb.NVMeNamespaceDe
 		return nil, err
 	}
 	log.Printf("Received from SPDK: %v", result)
+	delete (namespaces, namespace.Id.Value)
 	return &pb.NVMeNamespaceDeleteResponse{}, nil
 }
 
 func (s *server) NVMeNamespaceUpdate(ctx context.Context, in *pb.NVMeNamespaceUpdateRequest) (*pb.NVMeNamespaceUpdateResponse, error) {
-	log.Printf("Received from client: %v", in.GetNamespace())
+	log.Printf("Received from client: %v", in.Namespace)
+	namespaces[in.Namespace.Id.Value] = *in.Namespace
 	return &pb.NVMeNamespaceUpdateResponse{}, nil
 }
 
 func (s *server) NVMeNamespaceList(ctx context.Context, in *pb.NVMeNamespaceListRequest) (*pb.NVMeNamespaceListResponse, error) {
 	log.Printf("NVMeNamespaceList: Received from client: %v", in)
+	
+	nqn := "all"
```

Comment:
either `all` or empty `SubsystemId` can indicate we want everything

Reply:
ack; fixed.